compiler/rustc_passes/src/check_attr.rs

```diff
@@ -390,11 +390,33 @@ impl CheckAttrVisitor<'tcx> {
             .emit();
     }
 
-    fn check_doc_alias(&self, meta: &NestedMetaItem, hir_id: HirId, target: Target) -> bool {
-        let doc_alias = meta.value_str().map(|s| s.to_string()).unwrap_or_else(String::new);
+    fn check_doc_alias_value(
+        &self,
+        meta: &NestedMetaItem,
+        doc_alias: &str,
+        hir_id: HirId,
+        target: Target,
+        is_list: bool,
+    ) -> bool {
+        let err_fn = |tcx: TyCtxt<'_>, span: Span, msg: &str| {
```

Comment:
nit: `tcx` is always `self.tcx`, you don't need to make it a parameter. If you get borrow check errors you could use `let tcx = self.tcx` so you don't capture the whole struct.